pkgs/servers/pulseaudio/default.nix

```diff
@@ -45,6 +45,18 @@ stdenv.mkDerivation rec {
     # Install sysconfdir files inside of the nix store,
     # but use a conventional runtime sysconfdir outside the store
     ./add-option-for-installation-sysconfdir.patch
+
+    # https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/654
+    ./0001-Make-gio-2.0-optional-when-gsettings-is-disabled.patch
+
+    # TODO (not sent upstream)
+    ./0002-Ignore-SCM_CREDS-on-macOS.patch
+    ./0003-Disable-z-nodelete-on-darwin.patch
+    ./0004-Prefer-clock_gettime.patch
+    ./0005-Include-poll-posix.c-on-darwin.patch
+    ./0006-Only-use-version-script-on-GNU-ish-linkers.patch
+    ./0007-Adapt-undefined-link-args-per-linker.patch
+    ./0008-Use-correct-semaphore-on-darwin.patch
```

Comment:
I am not sure how I feel about all those non-upstreamed patches. For one thing they should only affect Darwin, but for other I feel like this will introduce a huge amount of burden until we upstream them (and I am not really interested in doing so, since I have no context).

I would propose that we make them conditional for Darwin and let it break on the next release (that is already cut in `master` but not released yet AFAIK), just to ship it to 22.05 and them eventually someone interested in Darwin can fix the patches later. WDYT?

BTW, my reasoning for this is that this package seems broken in `master`zfor a long time, and the only reason it got to our attention right now is the ZHF.

Reply:
Moved all those patches to be conditionally thanks to the reason above, and also to avoid a mass-rebuild in Linux.